cgi/suggest.pl

```diff
@@ -82,13 +82,17 @@
 		next if defined $just_synonyms{$tagtype}{$canon_tagid};
 		my $tag = $translations_to{$tagtype}{$canon_tagid}{$search_lc};
 		my $tagid = get_fileid($tag);
-		next if $tagid !~ /^$stringid/;
+		next if $tagid !~ /$stringid/;
 
 		if (not ($search_lc eq $original_lc)) {
 			$tag = $search_lc . ":" . $tag;
 		}
-
-		push @suggestions, $tag;
+		if ($tag =~ /^$stringid/) {
+			push @suggestions, $tag;
+		}
+		else {
+			unshift @suggestions, $tag;
```

Comment:
Isn't that going to put suggestions that do not begin with the string first, in reverse alphabetical order? It may be easier to have 2 arrays and to concatenate them after.

Reply:
Yeah that's right. I tried to keep just one array for performance but guess I'll implement your solution tomorrow.